Add custom frontend build

Netrvin opened the pull request:
This will allow developers to easily replace the frontend code with their own built versions, thereby more flexibly customizing the UI.

tpatel reviewed (changes requested):
Thanks for your contribution, this looks great!
I would love to see a test with a simple html file in the dist folders. This would help make sure the feature work in the long term, and help document the target location.

Netrvin commented:
> Thanks for your contribution, this looks great! I would love to see a test with a simple html file in the dist folders. This would help make sure the feature work in the long term, and help document the target location.

Ok, I have added a test. Thanks to this, I found a mistake in the configuration I added. Tests do matter.

tpatel reviewed (approved):
Thanks for your contribution ✨
